pinot-controller/src/main/java/com/linkedin/pinot/controller/util/TableSizeReader.java

```diff
@@ -194,26 +194,33 @@ public TableSubTypeSizeDetails getTableSubtypeSize(String tableNameWithType, int
       // after iterating over all servers update summary reported and estimated size of the segment
       if (errors != segmentSizes.serverInfo.size()) {
         // atleast one server reported size for this segment
+        LOGGER.info("Could not get size for segment {} from {} servers. Using segmentLevelMax {} to estimate the size",
```

Comment:
What happens if we segmentLevelMax is -1 (because we got errors from all replicas hosting a specific segment)? I know you mentioned there is an error in the math, if you are planning to fix it later, can u pls add a TODO?

Reply:
In that case, the code uses table-level max segment size. If table level max is -1, then we return -1 as the estimate.